src/mission/woper.jl

```diff
@@ -1,41 +1,37 @@
 """
-    woper(pari, parg, parm, para, pare, 
-          parad, pared, itermax, initeng, NPSS_PT, NPSS)
+    woper(ac, mi, itermax, initeng, saveOffDesign)
 
 `woper` runs the aircraft through input off-design missions
 
-!!! compat "Future Changes"
-      In an upcoming revision, an `aircraft` struct and auxiliary indices will be passed in lieu of pre-sliced `par` arrays.
+!!! details "🔃 Inputs and Outputs"
+**Inputs:**
+- `ac::aircraft`: Aircraft with first mission being the design mission
+- `mi::Int4`: Off design mission to run (Default: 1)
+- `itermax::Int64`: Maximum iterations for sizing loop
+- `initeng::Boolean`: Use design case as initial guess for engine state if true
+- `saveOffDesign::Boolean`: Set true if you want computed quanties to be saved in the selected off design par arrays of the aircraft model
+
+**Outputs:**
+- No explicit outputs. Computed quantities are saved to `par` arrays of `aircraft` model for the off design mission selected
 
 """
-function woper(pari, parg, parm, para, pare, 
-              parad, pared, itermax, initeng, NPSS_PT, NPSS)
+function woper(ac, mi = 1; itermax = 35, initeng = true, saveOffDesign = false)
 
-    # # Initialze some variables
-    # ifirst = true
-    # NPSS = Base.Process
-    # NPSS_TS = Base.Process
-    # NPSS_Fan = Base.Process
-    # NPSS_AftFan = Base.Process
-    # NPSS_PT = true
+    pari = ac.pari
+    parg = ac.parg
+    parm = ac.parm[:,mi:mi]
+    para = ac.para[:,:,mi:mi]
+    pare = ac.pare[:,:,mi:mi]
+    parad = ac.para[:,:,1:1]
```

Comment:
This will create copies of the array, making it slower. Could use the new `parad = ac.parad` which directly gives a view of the array. Similarly `ac.pared; ac.parmd` etc should work.